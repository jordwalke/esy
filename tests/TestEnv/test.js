const {createTestEnv} = require('../harness');

let sandbox = createTestEnv({sandbox: __dirname});

sandbox.exec(`
pushd PackageC
popd

pushd PackageB
rm -rf node_modules
mkdir node_modules
cd node_modules
ln -s ../../PackageC ./PackageC
popd

pushd PackageA
rm -rf node_modules
mkdir node_modules
cd node_modules
ln -s ../../PackageC ./PackageC
ln -s ../../PackageB ./PackageB
popd
`);

test('environment for PackageC', () => {
  let res = sandbox.exec(`
  cd PackageC
  ../../../.bin/esy
  `);
  expect(res.status).toBe(0);
  expect(res.stdout.toString()).toMatchSnapshot();
});

test('environment for PackageB', () => {
  let res = sandbox.exec(`
  cd PackageB
  ../../../.bin/esy
  `);
  expect(res.status).toBe(0);
  expect(res.stdout.toString()).toMatchSnapshot();
});

test('environment for PackageA', () => {
  let res = sandbox.exec(`
  cd PackageA
  ../../../.bin/esy
  `);
  expect(res.status).toBe(0);
  expect(res.stdout.toString()).toMatchSnapshot();
});
