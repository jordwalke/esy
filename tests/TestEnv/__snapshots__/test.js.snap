exports[`test environment for PackageA 1`] = `
"
# EsySandboxVariables 
export PATH=\"/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin\"
export SHELL=\"env -i /bin/bash --norc --noprofile\"
export esy__platform=\"platform\"
export esy__architecture=\"architecture\"
export esy__target_platform=\"platform\"
export esy__target_architecture=\"architecture\"
# [BuiltIns]
export esy__sandbox=\"$ESY__SANDBOX\"
export esy__store=\"$ESY__STORE\"
export esy__install_tree=\"$esy__sandbox/_install\"
export esy__build_tree=\"$esy__sandbox/_build\"
export cur__name=\"PackageA\"
export cur__version=\"1.0.0\"
export cur__root=\"$esy__sandbox/\"
export cur__depends=\"PackageB\"
export cur__target_dir=\"$esy__sandbox/_build\"
export cur__install=\"$esy__sandbox/_install\"
export cur__bin=\"$cur__install/bin\"
export cur__sbin=\"$cur__install/sbin\"
export cur__lib=\"$cur__install/lib\"
export cur__man=\"$cur__install/man\"
export cur__doc=\"$cur__install/doc\"
export cur__stublibs=\"$cur__install/stublibs\"
export cur__toplevel=\"$cur__install/toplevel\"
export cur__share=\"$cur__install/share\"
export cur__etc=\"$cur__install/etc\"
# [Custom Variables]
export OCAMLFIND_CONF=\"$cur__target_dir/_esy/findlib.conf\"
export PATH=\"$esy__store/_install/packageb-1.0.0/bin:$esy__store/_install/packagec-1.0.0/bin:$PATH\"
export MAN_PATH=\"$esy__store/_install/packageb-1.0.0/man:$esy__store/_install/packagec-1.0.0/man:$MAN_PATH\"

# PackageC@1.0.0 $esy__sandbox/../PackageC/package.json
# [BuiltIns]
export packagec__name=\"PackageC\"
export packagec__version=\"1.0.0\"
export packagec__root=\"$esy__sandbox/../PackageC\"
export packagec__depends=\"\"
export packagec__target_dir=\"$esy__store/_build/packagec-1.0.0\"
export packagec__install=\"$esy__store/_install/packagec-1.0.0\"
export packagec__bin=\"$packagec__install/bin\"
export packagec__sbin=\"$packagec__install/sbin\"
export packagec__lib=\"$packagec__install/lib\"
export packagec__man=\"$packagec__install/man\"
export packagec__doc=\"$packagec__install/doc\"
export packagec__stublibs=\"$packagec__install/stublibs\"
export packagec__toplevel=\"$packagec__install/toplevel\"
export packagec__share=\"$packagec__install/share\"
export packagec__etc=\"$packagec__install/etc\"
# [Custom Variables]
export packagec__dynamicVar=\"$cur__root\"
export packagec__LOCALVAR=\"$packagec__root/package.json\"
export GLOBAL_TEST_VAR_CLOBBER=\"$packagec__root/package.json\"
export GLOBAL_TEST_VAR_JOIN=\"$packagec__root/package.json:$GLOBAL_TEST_VAR_JOIN\"
export GLOBAL_TEST_VAR_PACKAGEC=\"$packagec__root/package.json\"

# PackageB@1.0.0 $esy__sandbox/../PackageB/package.json
# [ERROR] LOOKSLIKE__ANOTHER_PACKAGE looks like it\'s trying to step on another package because it has a double underscore - which is how we express namespaced env vars. The package owner for PackageB likely made a mistake
# [BuiltIns]
export packageb__name=\"PackageB\"
export packageb__version=\"1.0.0\"
export packageb__root=\"$esy__sandbox/../PackageB\"
export packageb__depends=\"PackageC\"
export packageb__target_dir=\"$esy__store/_build/packageb-1.0.0\"
export packageb__install=\"$esy__store/_install/packageb-1.0.0\"
export packageb__bin=\"$packageb__install/bin\"
export packageb__sbin=\"$packageb__install/sbin\"
export packageb__lib=\"$packageb__install/lib\"
export packageb__man=\"$packageb__install/man\"
export packageb__doc=\"$packageb__install/doc\"
export packageb__stublibs=\"$packageb__install/stublibs\"
export packageb__toplevel=\"$packageb__install/toplevel\"
export packageb__share=\"$packageb__install/share\"
export packageb__etc=\"$packageb__install/etc\"
# [Custom Variables]
export packageb__dynamicVar=\"$cur__root\"
export packageb__LOCALVAR=\"$packageb__root/package.json\"
export LOOKSLIKE__ANOTHER_PACKAGE=\"$packageb__root/package.json\"
export GLOBAL_TEST_VAR_CLOBBER=\"$packageb__root/package.json\"
export GLOBAL_TEST_VAR_JOIN=\"$packageb__root/package.json:$GLOBAL_TEST_VAR_JOIN\"
export GLOBAL_TEST_VAR_PACKAGEB=\"$packageb__root/package.json\"

# PackageA@1.0.0 $esy__sandbox/package.json
# [BuiltIns]
export packagea__name=\"PackageA\"
export packagea__version=\"1.0.0\"
export packagea__root=\"$esy__sandbox/\"
export packagea__depends=\"PackageB\"
export packagea__target_dir=\"$esy__sandbox/_build\"
export packagea__install=\"$esy__sandbox/_install\"
export packagea__bin=\"$packagea__install/bin\"
export packagea__sbin=\"$packagea__install/sbin\"
export packagea__lib=\"$packagea__install/lib\"
export packagea__man=\"$packagea__install/man\"
export packagea__doc=\"$packagea__install/doc\"
export packagea__stublibs=\"$packagea__install/stublibs\"
export packagea__toplevel=\"$packagea__install/toplevel\"
export packagea__share=\"$packagea__install/share\"
export packagea__etc=\"$packagea__install/etc\"
# [Custom Variables]
export packagea__dynamicVar=\"$cur__root\"
export packagea__LOCALVAR=\"$packagea__root/package.json\"
export GLOBAL_TEST_VAR_CLOBBER=\"$packagea__root/package.json\"
export GLOBAL_TEST_VAR_JOIN=\"$packagea__root/package.json:$GLOBAL_TEST_VAR_JOIN\"
export GLOBAL_TEST_VAR_PACKAGEA=\"$packagea__root/package.json\"
"
`;

exports[`test environment for PackageB 1`] = `
"
# EsySandboxVariables 
export PATH=\"/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin\"
export SHELL=\"env -i /bin/bash --norc --noprofile\"
export esy__platform=\"platform\"
export esy__architecture=\"architecture\"
export esy__target_platform=\"platform\"
export esy__target_architecture=\"architecture\"
# [BuiltIns]
export esy__sandbox=\"$ESY__SANDBOX\"
export esy__store=\"$ESY__STORE\"
export esy__install_tree=\"$esy__sandbox/_install\"
export esy__build_tree=\"$esy__sandbox/_build\"
export cur__name=\"PackageB\"
export cur__version=\"1.0.0\"
export cur__root=\"$esy__sandbox/\"
export cur__depends=\"PackageC\"
export cur__target_dir=\"$esy__sandbox/_build\"
export cur__install=\"$esy__sandbox/_install\"
export cur__bin=\"$cur__install/bin\"
export cur__sbin=\"$cur__install/sbin\"
export cur__lib=\"$cur__install/lib\"
export cur__man=\"$cur__install/man\"
export cur__doc=\"$cur__install/doc\"
export cur__stublibs=\"$cur__install/stublibs\"
export cur__toplevel=\"$cur__install/toplevel\"
export cur__share=\"$cur__install/share\"
export cur__etc=\"$cur__install/etc\"
# [Custom Variables]
export OCAMLFIND_CONF=\"$cur__target_dir/_esy/findlib.conf\"
export PATH=\"$esy__store/_install/packagec-1.0.0/bin:$PATH\"
export MAN_PATH=\"$esy__store/_install/packagec-1.0.0/man:$MAN_PATH\"

# PackageC@1.0.0 $esy__sandbox/../PackageC/package.json
# [BuiltIns]
export packagec__name=\"PackageC\"
export packagec__version=\"1.0.0\"
export packagec__root=\"$esy__sandbox/../PackageC\"
export packagec__depends=\"\"
export packagec__target_dir=\"$esy__store/_build/packagec-1.0.0\"
export packagec__install=\"$esy__store/_install/packagec-1.0.0\"
export packagec__bin=\"$packagec__install/bin\"
export packagec__sbin=\"$packagec__install/sbin\"
export packagec__lib=\"$packagec__install/lib\"
export packagec__man=\"$packagec__install/man\"
export packagec__doc=\"$packagec__install/doc\"
export packagec__stublibs=\"$packagec__install/stublibs\"
export packagec__toplevel=\"$packagec__install/toplevel\"
export packagec__share=\"$packagec__install/share\"
export packagec__etc=\"$packagec__install/etc\"
# [Custom Variables]
export packagec__dynamicVar=\"$cur__root\"
export packagec__LOCALVAR=\"$packagec__root/package.json\"
export GLOBAL_TEST_VAR_CLOBBER=\"$packagec__root/package.json\"
export GLOBAL_TEST_VAR_JOIN=\"$packagec__root/package.json:$GLOBAL_TEST_VAR_JOIN\"
export GLOBAL_TEST_VAR_PACKAGEC=\"$packagec__root/package.json\"

# PackageB@1.0.0 $esy__sandbox/package.json
# [ERROR] LOOKSLIKE__ANOTHER_PACKAGE looks like it\'s trying to step on another package because it has a double underscore - which is how we express namespaced env vars. The package owner for PackageB likely made a mistake
# [BuiltIns]
export packageb__name=\"PackageB\"
export packageb__version=\"1.0.0\"
export packageb__root=\"$esy__sandbox/\"
export packageb__depends=\"PackageC\"
export packageb__target_dir=\"$esy__sandbox/_build\"
export packageb__install=\"$esy__sandbox/_install\"
export packageb__bin=\"$packageb__install/bin\"
export packageb__sbin=\"$packageb__install/sbin\"
export packageb__lib=\"$packageb__install/lib\"
export packageb__man=\"$packageb__install/man\"
export packageb__doc=\"$packageb__install/doc\"
export packageb__stublibs=\"$packageb__install/stublibs\"
export packageb__toplevel=\"$packageb__install/toplevel\"
export packageb__share=\"$packageb__install/share\"
export packageb__etc=\"$packageb__install/etc\"
# [Custom Variables]
export packageb__dynamicVar=\"$cur__root\"
export packageb__LOCALVAR=\"$packageb__root/package.json\"
export LOOKSLIKE__ANOTHER_PACKAGE=\"$packageb__root/package.json\"
export GLOBAL_TEST_VAR_CLOBBER=\"$packageb__root/package.json\"
export GLOBAL_TEST_VAR_JOIN=\"$packageb__root/package.json:$GLOBAL_TEST_VAR_JOIN\"
export GLOBAL_TEST_VAR_PACKAGEB=\"$packageb__root/package.json\"
"
`;

exports[`test environment for PackageC 1`] = `
"
# EsySandboxVariables 
export PATH=\"/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin\"
export SHELL=\"env -i /bin/bash --norc --noprofile\"
export esy__platform=\"platform\"
export esy__architecture=\"architecture\"
export esy__target_platform=\"platform\"
export esy__target_architecture=\"architecture\"
# [BuiltIns]
export esy__sandbox=\"$ESY__SANDBOX\"
export esy__store=\"$ESY__STORE\"
export esy__install_tree=\"$esy__sandbox/_install\"
export esy__build_tree=\"$esy__sandbox/_build\"
export cur__name=\"PackageC\"
export cur__version=\"1.0.0\"
export cur__root=\"$esy__sandbox/\"
export cur__depends=\"\"
export cur__target_dir=\"$esy__sandbox/_build\"
export cur__install=\"$esy__sandbox/_install\"
export cur__bin=\"$cur__install/bin\"
export cur__sbin=\"$cur__install/sbin\"
export cur__lib=\"$cur__install/lib\"
export cur__man=\"$cur__install/man\"
export cur__doc=\"$cur__install/doc\"
export cur__stublibs=\"$cur__install/stublibs\"
export cur__toplevel=\"$cur__install/toplevel\"
export cur__share=\"$cur__install/share\"
export cur__etc=\"$cur__install/etc\"
# [Custom Variables]
export OCAMLFIND_CONF=\"$cur__target_dir/_esy/findlib.conf\"

# PackageC@1.0.0 $esy__sandbox/package.json
# [BuiltIns]
export packagec__name=\"PackageC\"
export packagec__version=\"1.0.0\"
export packagec__root=\"$esy__sandbox/\"
export packagec__depends=\"\"
export packagec__target_dir=\"$esy__sandbox/_build\"
export packagec__install=\"$esy__sandbox/_install\"
export packagec__bin=\"$packagec__install/bin\"
export packagec__sbin=\"$packagec__install/sbin\"
export packagec__lib=\"$packagec__install/lib\"
export packagec__man=\"$packagec__install/man\"
export packagec__doc=\"$packagec__install/doc\"
export packagec__stublibs=\"$packagec__install/stublibs\"
export packagec__toplevel=\"$packagec__install/toplevel\"
export packagec__share=\"$packagec__install/share\"
export packagec__etc=\"$packagec__install/etc\"
# [Custom Variables]
export packagec__dynamicVar=\"$cur__root\"
export packagec__LOCALVAR=\"$packagec__root/package.json\"
export GLOBAL_TEST_VAR_CLOBBER=\"$packagec__root/package.json\"
export GLOBAL_TEST_VAR_JOIN=\"$packagec__root/package.json:$GLOBAL_TEST_VAR_JOIN\"
export GLOBAL_TEST_VAR_PACKAGEC=\"$packagec__root/package.json\"
"
`;
